ansible/roles/gelato-ha-installer/tasks/main.yml

```diff
@@ -0,0 +1,78 @@
+# Copyright 2021 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Note: It is assumed that the Kubernetes master node has all the prereqs for running Multicloud services. 
+
+---
+- name: create gelato-ha work directory if it doesn't exist
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0755
+  with_items:
+    - "{{ gelato_work_dir }}"
+
+- name: include scenarios/release.yml when installing from release
+  include: scenarios/release.yml
+  when: install_gelato_ha_from == "release" or install_gelato_ha_from == "container" or install_gelato_ha_from == "repository"
+
+- name: Update the api, file, s3 deployments
+  replace: dest="{{ gelato_work_dir }}/multi-cloud/K8s/{{item}}" regexp='KEYSTONE_IP' replace='{{ host_ip }}'
+  become: yes
+  with_items:
+    - api-deployment.yaml
+    - file-deployment.yaml
+    - s3api-deployment.yaml
+  when: 
+    - hostvars["controller"]["controller_ip"] is defined
+
+- name: Execute the kubectl yamls.
+  shell: "{{ item }}"
+  args:
+    chdir: "{{ gelato_work_dir }}/multi-cloud/K8s"
+  with_items:
+    - bash "k8s-mongo.sh"
+    - bash "k8s-deploy.sh"
+  become_user: ubuntu
+  when: install_gelato_ha_from == "release" or install_gelato_ha_from == "container" or install_gelato_ha_from == "repository"
+
+  # Get the list of servicses deployed.
+- name: get multi-cloud service status
+  shell: kubectl get all -n=soda-multi-cloud
+  become_user: ubuntu
+
+  
+- name: get the IPs of API Service.
+  shell: kubectl get nodes k8smaster -o wide --no-headers | awk '{print $6}'
+  register: api_ip
+  become_user: ubuntu
+  when: install_gelato_ha_from == "release" or install_gelato_ha_from == "container" or install_gelato_ha_from == "repository"
```

Comment:
Why this condition? what condition is excluded?

Reply:
for any condition, use only the supported repository method of installing HA

Reply:
In that case check for 'when' can be removed